Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+            SceneLoadedForPlayer(conn, conn.playerController.gameObject);
+        }
+
+        internal void ReadyStatusChanged()
+        {
+            int CurrentPlayers = 0;
+            int ReadyPlayers = 0;
+            foreach (NetworkLobbyPlayer item in lobbySlots)
+            {
+                if (item != null)
+                {
+                    CurrentPlayers++;
+                    if (item.ReadyToBegin)
+                        ReadyPlayers++;
+                }
+            }
+            if (CurrentPlayers == ReadyPlayers)
+                CheckReadyToBegin();
+            else
+                allPlayersReady = false;
+        }
+
+        void SceneLoadedForPlayer(NetworkConnection conn, GameObject lobbyPlayerGameObject)
+        {
+            // if not a lobby player.. dont replace it
+            if (lobbyPlayerGameObject.GetComponent<NetworkLobbyPlayer>() == null) return;
+
+            if (LogFilter.Debug) Debug.LogFormat("NetworkLobby SceneLoadedForPlayer scene: {0} {1}", SceneManager.GetActiveScene().name, conn);
+
+            if (SceneManager.GetActiveScene().name == LobbyScene)
+            {
+                // cant be ready in lobby, add to ready list
+                PendingPlayer pending;
+                pending.conn = conn;
+                pending.lobbyPlayer = lobbyPlayerGameObject;
+                pendingPlayers.Add(pending);
+                return;
+            }
+
+            GameObject gamePlayer = OnLobbyServerCreateGamePlayer(conn);
+            if (gamePlayer == null)
+            {
+                // get start position from base class
+                Transform startPos = GetStartPosition();
+                gamePlayer = startPos != null
+                    ? Instantiate(playerPrefab, startPos.position, startPos.rotation)
+                    : Instantiate(playerPrefab, Vector3.zero, Quaternion.identity);
+                gamePlayer.name = playerPrefab.name;
+            }
+
+            if (!OnLobbyServerSceneLoadedForPlayer(lobbyPlayerGameObject, gamePlayer))
+                return;
+
+            // replace lobby player with game player
+            NetworkServer.ReplacePlayerForConnection(conn, gamePlayer);
+        }
+
+        static int CheckConnectionIsReadyToBegin(NetworkConnection conn)
+        {
+            int countPlayers = 0;
+            if (conn.playerController.gameObject.GetComponent<NetworkLobbyPlayer>().ReadyToBegin)
+                countPlayers += 1;
+
+            return countPlayers;
+        }
+
+        public void CheckReadyToBegin()
+        {
+            if (SceneManager.GetActiveScene().name != LobbyScene) return;
+
+            int readyCount = 0;
+
+            foreach (KeyValuePair<int, NetworkConnection> conn in NetworkServer.connections)
+            {
+                if (conn.Value == null)
+                    continue;
+
+                readyCount += CheckConnectionIsReadyToBegin(conn.Value);
```

Comment:
please use Linq to make this less magic. readyCount = networkserver.connections.count(conn => conn.value != null && CheckConnectionIsReadyToBegin(conn.value));

Reply:
Don't ever use LINQ, it creates garbage, which is a big problem in the unity mono.

Reply:
@Vincenz099 This method is called once per player as they join and click a "Ready" button...LINQ is fine here.

Reply:
Done